src/snowflake/snowpark/session.py

```diff
@@ -870,81 +896,73 @@ def _resolve_packages(
         if not self.get_current_database():
             package_table = f"snowflake.{package_table}"
 
+        package_search_list = [v[0] for v in package_dict.values()] + [
+            SNOWPARK_PACKAGE_NAME
+        ]
         valid_packages = (
             {
                 p[0]: json.loads(p[1])
                 for p in self.table(package_table)
                 .filter(
                     (col("language") == "python")
-                    & (col("package_name").in_([v[0] for v in package_dict.values()]))
+                    & (col("package_name").in_(package_search_list))
                 )
                 .group_by("package_name")
                 .agg(array_agg("version"))
                 ._internal_collect_with_tag()
             }
-            if validate_package and package_dict
+            if len(package_search_list) > 0
```

Comment:
we should keep `validate_package` here

Reply:
It was removed because we need the `valid_packages` list to have access to the versions for snowpark package (because it was being added automatically). The `validate_package` check prevents validation later in the code, so is redundant here.

I have reverted this to match earlier code as we are no longer automatically adding the snowpark package